### DISCUSSION TITLE
Simulate B-mode US images

### DISCUSSION THREAD

**Discussion by kwsp** [ucl-bug/jwave#104] [Ideas]
Are there examples for simulating B-mode US images like k-wave? For example, http://www.k-wave.org/documentation/example_us_bmode_linear_transducer.php

It seems like the capability is already there - just need to implement a transducer class to facilitate easier simulation. 

Any help is appreciated! Thanks you!

**Comment by astanziola** [comment]
Hi @kwsp !
Yes it is fairly easy to make a B-Mode image. Probably the only real work to do is indeed to extend the transducer class to define the transmit and beamforming schemes. 

The most important issue (which is shared with [FWI algorithms](https://library.seg.org/doi/10.1190/geo2017-0063.1)) is the large memory requirements in 3D for backpropagation. In practice, one needs to store the entire wavefield at every single timestep of the simulation to accurately calculate the gradients.

But it is still possible to make a 2D example which can be easily extended to 3D. I will make an issue to do so 😄

**Reply by rohandekate10** [reply]
Is there a tutorial available for this?
